### PR TITLE
Fixed C++ BatchNorm pretty_print() with optional momentum

### LINKED ISSUE #65143: C++ BatchNorm pretty_print() fails with optional momentum
## 🐛 Bug

In _nn/modules/batchnorm.cpp,_
the _pretty_print_ method does not handle the case when the momentum option is not defined:

```
template <size_t D, typename Derived>
void BatchNormImplBase<D, Derived>::pretty_print(std::ostream& stream) const {
  stream << std::boolalpha
         << "torch::nn::BatchNorm" << D << "d("
         << this->options.num_features() << ", "
         << "eps=" << this->options.eps() << ", "
         << "momentum=" << this->options.momentum().value() << ", "
         << "affine=" << this->options.affine() << ", "
         << "track_running_stats=" << this->options.track_running_stats() << ")";
}
```

## To Reproduce

```
torch::nn::BatchNorm2d m(torch::nn::BatchNormOptions(128).momentum(c10::nullopt));
std::cerr << *m << "\n";

torch::nn::BatchNorm2d(128, eps=1e-05, momentum=terminate called after throwing an instance of 'c10::bad_optional_access'
  what():  bad optional access
```

## Expected behavior

From the python side, the momentum is printed as "None"

```
>>> m=torch.nn.BatchNorm2d(128,momentum=None)
>>> print(m)
BatchNorm2d(128, eps=1e-05, momentum=None, affine=True, track_running_stats=True)
```

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @albanD @mruberry @walterddr

### PR DESCRIPTION
**Pull request by francescocastelli**
Summary : Inserted a check for the momentum and print  "None" in case is not defined. See  #65143

Test: The code below now prints `torch::nn::BatchNorm2d(128, eps=1e-05, momentum=None, affine=true, track_running_stats=true)` without generating errors.
```
torch::nn::BatchNorm2d m(torch::nn::BatchNormOptions(128).momentum(c10::nullopt));
std::cerr << *m << "\n";
```
Fixes #65143

### CODE REVIEW

**Review by ngimel — APPROVED** [review]

**Comment by francescocastelli** [comment]
@ngimel Done!
